src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -649,6 +649,23 @@ public function dumpFile($filename, $content)
     }
 
     /**
+     * Atomically appends content to an existing file.
+     *
+     * @param string $filename The file for which to append content
+     * @param string $content  The content to append
+     *
+     * @throws IOException If the file is not writable
+     */
+    public function appendToFile($filename, $content)
+    {
+        if (false !== $originalContent = @file_get_contents($filename)) {
```

Comment:
This now loads the current file content into memory (which doesn't look like an atomic append to me). Wouldn't it be better to rely on the `FILE_APPEND` flag of `file_put_contents` instead? 
You could refactor a bit the logic inside `dumpFile` in order to extract it in a private method adding the flag or not.